test/ex_machina/ecto_strategy_test.exs

```diff
@@ -59,6 +59,19 @@ defmodule ExMachina.EctoStrategyTest do
     assert List.first(model.articles).visits == Decimal.new(10)
   end
 
+  test "insert/1 casts embedded associations" do
+    author = %ExMachina.Author{name: "Paul", salary: 10.3}
+    link = %ExMachina.Link{url: "wow"}
+
+    comment = TestFactory.insert :comment_with_embedded_assocs,
+      author: author,
+      links: [link]
+
+    assert comment.author.name == author.name
+    assert comment.author.salary == author.salary
```

Comment:
Derp. Thanks. That means it probably isn't casting. I'll fix the test and add the corresponding code